diff --git a/config/schema/geolocation.schema.yml b/config/schema/geolocation.schema.yml
index dabe4f9..f8dbfe8 100644
--- a/config/schema/geolocation.schema.yml
+++ b/config/schema/geolocation.schema.yml
@@ -115,6 +115,9 @@ views.style.maps_common:
     marker_scroll_to_result:
       type: integer
       label: 'Scroll instead of open bubble'
+    marker_row_number:
+      type: integer
+      label: 'Use row number as marker label'
     id_field:
       type: string
       label: 'ID field'
diff --git a/geolocation.module b/geolocation.module
index 63d4da7..6346789 100644
--- a/geolocation.module
+++ b/geolocation.module
@@ -75,6 +75,7 @@ function geolocation_theme() {
         'position' => NULL,
         'icon' => NULL,
         'location_id' => NULL,
+        'marker_label' => NULL,
       ],
     ],
   ];
diff --git a/js/geolocation-common-map.js b/js/geolocation-common-map.js
index 5cae68f..8309e83 100644
--- a/js/geolocation-common-map.js
+++ b/js/geolocation-common-map.js
@@ -374,6 +374,10 @@
             markerConfig.icon = location.data('icon');
           }
 
+          if (typeof location.data('markerLabel') !== 'undefined') {
+            markerConfig.label = location.data('markerLabel').toString();
+          }
+
           var skipInfoWindow = false;
           if (commonMapSettings.markerScrollToResult === true) {
             skipInfoWindow = true;
diff --git a/js/geolocation-google-maps-api.js b/js/geolocation-google-maps-api.js
index 73f71f9..7fbbf2a 100644
--- a/js/geolocation-google-maps-api.js
+++ b/js/geolocation-google-maps-api.js
@@ -88,6 +88,7 @@
  * @property {GoogleMap} map
  * @property {string} title
  * @property {string} [icon]
+ * @property {string} [label]
  *
  * Settings from Geolocation module:
  * @property {string} [infoWindowContent]
diff --git a/src/Plugin/views/style/CommonMap.php b/src/Plugin/views/style/CommonMap.php
index dd7b402..d892ba5 100644
--- a/src/Plugin/views/style/CommonMap.php
+++ b/src/Plugin/views/style/CommonMap.php
@@ -200,7 +200,7 @@ class CommonMap extends StylePluginBase {
     /*
      * Add locations to output.
      */
-    foreach ($this->view->result as $row) {
+    foreach ($this->view->result as $row_number => $row) {
       if (!empty($title_field)) {
         $title_field_handler = $this->view->field[$title_field];
         $title_build = [
@@ -272,6 +272,9 @@ class CommonMap extends StylePluginBase {
         if (!empty($location_id)) {
           $location['#location_id'] = $location_id;
         }
+        if ($this->options['marker_row_number']) {
+          $location['#marker_label'] = (int) $row_number + 1;
+        }
 
         $build['#locations'][] = $location;
       }
@@ -407,6 +410,7 @@ class CommonMap extends StylePluginBase {
     $options['title_field'] = ['default' => ''];
     $options['icon_field'] = ['default' => ''];
     $options['marker_scroll_to_result'] = ['default' => 0];
+    $options['marker_row_number'] = ['default' => FALSE];
     $options['id_field'] = ['default' => ''];
     $options['marker_clusterer'] = ['default' => 0];
     $options['marker_clusterer_image_path'] = ['default' => ''];
@@ -745,6 +749,13 @@ class CommonMap extends StylePluginBase {
       ];
     }
 
+    $form['marker_row_number'] = [
+      '#group' => 'style_options][advanced_settings',
+      '#title' => $this->t('Show views result row number in marker'),
+      '#type' => 'checkbox',
+      '#default_value' => $this->options['marker_row_number'],
+    ];
+
     $form['context_popup_content'] = [
       '#group' => 'style_options][advanced_settings',
       '#type' => 'textarea',
diff --git a/templates/geolocation-common-map-location.html.twig b/templates/geolocation-common-map-location.html.twig
index de82bd8..fdecef7 100644
--- a/templates/geolocation-common-map-location.html.twig
+++ b/templates/geolocation-common-map-location.html.twig
@@ -7,7 +7,7 @@
 
   Changing the HTML tags, adding classes or adding content around or within the existing structure is no problem.
 #}
-<div class="geolocation" data-lat="{{ position.lat }}" data-lng="{{ position.lng }}" {% if icon is not empty %} data-icon="{{ icon }}" {% endif %} {% if location_id is not empty %} data-location-id="{{ location_id }}" {% endif %}>
+<div class="geolocation" data-lat="{{ position.lat }}" data-lng="{{ position.lng }}" {% if icon is not empty %} data-icon="{{ icon }}" {% endif %} {% if location_id is not empty %} data-location-id="{{ location_id }}" {% endif %} {% if marker_label is not empty %} data-marker-label="{{ marker_label }}" {% endif %}>
     <h2 class="location-title">{{ title }}</h2>
     <div class="location-content">{{ content }}</div>
 </div>
